backends/gstreamer/src/media_capture.rs

```diff
@@ -10,16 +9,22 @@ pub enum Constrain<T> {
 }
 
 impl Constrain<u64> {
-    fn add_to_caps(self, name: &str, min: u64, max: u64, builder: Builder) -> Option<Builder> {
+    fn add_to_caps(
+        &self,
+        name: &str,
+        min: u64,
+        max: u64,
+        builder: gst::caps::Builder,
```

Comment:
You need to specify the lifetime parameter of the builder here and in the return value. And then it either works or you're violating the lifetime requirements in the body of the function :)

Reply:
thanks!